test/e2e/e2e_setup_test.go

```diff
@@ -185,23 +144,126 @@ func Setup(specName string, f func(testSpecificSettings func() testSettings), op
+							Name:      testSpecificIPAddressClaims[0].Name,
+						},
+						// NOTE: we are omitting VMOperatorDependencies because it is not created yet (it will be created by the PostNamespaceCreated hook)
+						// But this is not a issue because a default dependenciesConfig that works for vcsim will be automatically used.
+					},
+				}
+
+				err = managementClusterProxy.GetClient().Create(ctx, envVar)
+				Expect(err).ToNot(HaveOccurred(), "Failed to create EnvVar")
+
+				Eventually(func() bool {
+					if err := managementClusterProxy.GetClient().Get(ctx, crclient.ObjectKeyFromObject(envVar), envVar); err != nil {
+						return false
+					}
+					return len(envVar.Status.Variables) > 0
+				}, 30*time.Second, 5*time.Second).Should(BeTrue(), "Failed to get EnvVar %s", klog.KObj(envVar))
+
+				Byf("Setting test variables for %s", specName)
+				for k, v := range envVar.Status.Variables {
+					// ignore variables that will be set later on by the test
+					if sets.New("NAMESPACE", "CLUSTER_NAME", "KUBERNETES_VERSION", "CONTROL_PLANE_MACHINE_COUNT", "WORKER_MACHINE_COUNT").Has(k) {
+						continue
+					}
+
+					// unset corresponding env variable (that in CI contains VMC data), so we are sure we use the value for vcsim
+					if strings.HasPrefix(k, "VSPHERE_") {
+						Expect(os.Unsetenv(k)).To(Succeed())
+					}
+
+					testSpecificVariables[k] = v
+				}
+			}
+
+			// Re-write the clusterctl config file and add the new variables.
+			testSpecificClusterctlConfigPath = fmt.Sprintf("%s-%s.yaml", strings.TrimSuffix(clusterctlConfigPath, ".yaml"), specName)
+			Byf("Writing a new clusterctl config to %s", testSpecificClusterctlConfigPath)
+			copyAndAmendClusterctlConfig(ctx, copyAndAmendClusterctlConfigInput{
+				ClusterctlConfigPath: testSpecificClusterctlConfigPath,
+				OutputPath:           testSpecificClusterctlConfigPath,
+				Variables:            testSpecificVariables,
+			})
+
+			// Run additional initialization required for supervisor.
+			if testMode == SupervisorTestMode {
+				switch testTarget {
+				case VCenterTestTarget:
+					setupNamespaceWithVMOperatorDependenciesVCenter(managementClusterProxy, workloadClusterNamespace)
+				case VCSimTestTarget:
+					setupNamespaceWithVMOperatorDependenciesVCSim(managementClusterProxy, workloadClusterNamespace)
+				}
+			}
 		}
 	})
 	defer AfterEach(func() {
-		Byf("Cleaning up test env for %s", specName)
-		switch testTarget {
-		case VCenterTestTarget:
-			// cleanup IPs/controlPlaneEndpoint created by the in cluster ipam provider.
-			Expect(inClusterAddressManager.Cleanup(ctx, testSpecificIPAddressClaims)).To(Succeed())
-		case VCSimTestTarget:
-			// cleanup IPs/controlPlaneEndpoint created by the vcsim controller manager.
-			Expect(vcsimAddressManager.Cleanup(ctx, testSpecificIPAddressClaims)).To(Succeed())
+		if !skipCleanup {
+			Byf("Cleaning up test env for %s", specName)
+			// We can't cleanup when a kind management cluster is used, because it won't exist anymore.
```

Comment:
> because it won't exist anymore.

Why?

I assume because the test itself cleans it up? Please document

(This seems very specific to the clusterctl upgrade test, right? not necessarily in general when we would set additionalVCSimServer for other tests in the future)